### PR TITLE
refactor: move dynamo/TorchXLA bridge to pytorch/xla repo

### PR DESCRIPTION
**Pull request by shunting314**
Check the pytorch side PR for details: https://github.com/pytorch/pytorch/pull/92601 . 

The pytorch one needs to wait for this one to be merged first.

### CODE REVIEW

**Comment by shunting314** [comment]
@JackCaoG  should I be worried about the CI break?
1. for the lint error, it looks like xla/ repo wants 2 spaces indent while pytorch/ repo usually uses 4. I can run some quick 'sed' command to format the code. Or let me know if there is already some script in the repo that can do the job.
2. the build error seems to be caused by some unrelated network error.

**Comment by JackCaoG** [comment]
I just restarted the CI, let's see how it goes. I think you also want to update the test under `test/dynamo` since the bridge is under a new path.

**Comment by shunting314** [comment]
Looks like the exsting tests under test/dynamo does not refer to the bridge directly. They specify the backend by string name which is not affected by the refactoring.

**Comment by wconstab** [comment]
should I review the code in this PR as if it is just a code move from pytorch/pytorch or is there also some significant change to the code?

**Comment by shunting314** [comment]
@wconstab  there is no significant change. Just some code movements and some necessary renames.

**Comment by JackCaoG** [comment]
Thanks Shutning, I will try to take a look tmr.

File: test/dynamo/test_bridge.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
+import copy
```
Comment (JackCaoG) [inline test/dynamo/test_bridge.py:1]: I assume this test is moved from upstram `test_torchxla_integration.py`. Let's replace 
https://github.com/pytorch/xla/blob/aff3513234fe344bbf159af5e376e5a45792f5dc/test/run_tests.sh#L135

with the new path then.

File: test/dynamo/test_num_output.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+import unittest
```
Comment (JackCaoG) [inline test/dynamo/test_num_output.py:1]: ditto, let's add this test to the `run_test.sh`

File: torch_xla/csrc/tensor.cpp [hunk 3]
```diff
@@ -702,12 +702,12 @@ c10::SymNode XLASymNodeImpl::neg() {
                    << " has not been implemented.";
 }
 
-c10::SymNode XLASymNodeImpl::min(const c10::SymNode& other) {
+c10::SymNode XLASymNodeImpl::sym_min(const c10::SymNode& other) {
```
Comment (JackCaoG) [inline torch_xla/csrc/tensor.cpp:705]: if you rebase, these file should be updated on master already

File: torch_xla/dynamo_bridge.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
+import dataclasses
```
Comment (JackCaoG) [inline torch_xla/dynamo_bridge.py:1]: let's move this file into `torch_xla/core` maybe? we generally don't have python file under `torch_xla` dir directly.

**Review by JackCaoG — COMMENTED** [review]
Mostly LGTM, thanks Shunting!

**Review by JackCaoG — APPROVED** [review]
Let's fix the linter then we can merge this one. Thanks @shunting314 !

**Comment by JackCaoG** [comment]
You can follow https://github.com/pytorch/xla/blob/master/CONTRIBUTING.md#before-submiting-a-pull-request

**Comment by JackCaoG** [comment]
failure is unrealted, merging

**Pull request merged**